src/plots/cartesian/select.js

```diff
@@ -221,6 +225,7 @@ module.exports = function prepSelect(e, startX, startY, dragOptions, mode) {
                     searchInfo.selectPoints(searchInfo, false);
                 }
 
+                updateSelectedState(gd, searchTraces);
                 gd.emit('plotly_deselect', null);
```

Comment:
@alexcjohnson The failing test on CI

![image](https://user-images.githubusercontent.com/6675409/32396579-29077bc0-c0bc-11e7-881f-b1a0e0b75df4.png)

can be fixed by moving `updateSelectedState` after `gd.emit`. I don't think this is desired behavior though. We should emit `plotly_selected` and `plotly_deselect` after `module.style()` is completed.

So, there must a race condition somewhere in `select_test.js`. Maybe we're missing a `Lib.clearThrottle` call?

Reply:
Shamefully fixed via https://github.com/plotly/plotly.js/pull/2135/commits/6437081d71ccaa5d144d9ee63b327625f643509e